tests/node_parse_code_subclass.phpt

```diff
@@ -0,0 +1,21 @@
+--TEST--
+Node::parseCode() on valid code with subclass
+--FILE--
+<?php
+
+require __DIR__ . '/../util.php';
+
+class MyNode extends \ast\Node {
+}
+
+$code = <<<'PHP'
+<?php
+
+return 123;
+PHP;
+
+echo get_class(MyNode::parseCode($code, $version=50)), "\n";
```

Comment:
1. it'd be more useful to test properties are correctly set, e.g. with the node dumper
2. could check that additional properties on the MyNode can be declared and that when read, they have the expected value. Same for properties inherited from traits (to have the unit test in case the property slot order ever changes and affects ast_node_set_prop_kind)